services/apigateway/src/main/java/software/amazon/awssdk/services/apigateway/internal/AcceptJsonInterceptor.java

```diff
@@ -26,9 +26,18 @@
     @Override
     public SdkHttpRequest modifyHttpRequest(Context.ModifyHttpRequest context,
                                             ExecutionAttributes executionAttributes) {
-        return context.httpRequest()
-                      .toBuilder()
-                      .putHeader("Accept", "application/json")
-                      .build();
+        // Some APIG operations marshall to the 'Accept' header to specify the
+        // format of the document returned by the service, such as GetExport
+        // https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-export-api.html.
+        // See the same fix in V1:
+        // https://github.com/aws/aws-sdk-java/blob/cd2275c07df8656033bfa9baa665354bfb17a6bf/aws-java-sdk-api-gateway/src/main/java/com/amazonaws/services/apigateway/internal/AcceptJsonRequestHandler.java#L29
```

Comment:
Another reason for this change is that customer can provide the header via `overrideConfiguration` and we should honor it :)